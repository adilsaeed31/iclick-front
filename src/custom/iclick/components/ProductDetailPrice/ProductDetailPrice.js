import React, { Component } from "react";
import PropTypes from "prop-types";
import Typography from "@material-ui/core/Typography";
import classnames from "classnames";
import Grid from "@material-ui/core/Grid";

class ProductDetailPrice extends Component {
  static propTypes = {
    /**
     * Sale or adjusted price.
     * If defined, the priceRange will be crossed out
     */
    compareAtPrice: PropTypes.string,

    /**
     * Use the compact layout
     */
    isCompact: PropTypes.bool,

    /**
     * Price or price range as a string
     */
    price: PropTypes.string
  }

  static defaultProps = {
    isCompact: false
  }

  render() {
    const { compareAtPrice, price, isCompact } = this.props;

    // If all props are undefined then skip rendering component
    if (!price) return null;

    if (isCompact) {
      return (
        <Grid item sm={12}>
          <Typography component="span" variant="body1">
            {price}
          </Typography>
          <Typography variant="caption">{compareAtPrice}</Typography>
        </Grid>
      );
    }

    return (
      <Grid item sm={12}>
        <Typography variant="caption">{compareAtPrice}</Typography>
        <Typography component="div" variant="title">
          {price}
        </Typography>
      </Grid>
    );
  }
}

export default ProductDetailPrice;
