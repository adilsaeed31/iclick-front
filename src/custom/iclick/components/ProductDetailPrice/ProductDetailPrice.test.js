import React from "react";
import renderer from "react-test-renderer";
import ProductDetailPrice from "./ProductDetailPrice";

test("price", () => {
  const component = renderer.create((
    <ProductDetailPrice price="$19.99" />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("no price", () => {
  const component = renderer.create((
    <ProductDetailPrice />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("no price, compareAtPrice", () => {
  const component = renderer.create((
    <ProductDetailPrice compareAtPrice="$29.99" />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("price and compareAtPrice", () => {
  const component = renderer.create((
    <ProductDetailPrice compareAtPrice="$29.99" price="$19.99" />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
