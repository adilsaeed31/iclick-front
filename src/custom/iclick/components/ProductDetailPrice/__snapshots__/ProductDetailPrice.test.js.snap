// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no price 1`] = `null`;

exports[`no price, compareAtPrice 1`] = `null`;

exports[`price 1`] = `
<div
  className="MuiGrid-item-7 MuiGrid-grid-sm-12-60 SKProductDetailPrice-root-1"
>
  <span
    className="MuiTypography-root-103 MuiTypography-caption-113 SKProductDetailPrice-strike-5"
  />
  <div
    className="MuiTypography-root-103 MuiTypography-title-109"
  >
    $19.99
  </div>
</div>
`;

exports[`price and compareAtPrice 1`] = `
<div
  className="MuiGrid-item-7 MuiGrid-grid-sm-12-60 SKProductDetailPrice-root-1"
>
  <span
    className="MuiTypography-root-103 MuiTypography-caption-113 SKProductDetailPrice-strike-5"
  >
    $29.99
  </span>
  <div
    className="MuiTypography-root-103 MuiTypography-title-109"
  >
    $19.99
  </div>
</div>
`;
